channel/app.go

```diff
@@ -163,6 +172,33 @@ func (d OptAppDec) Decode(r io.Reader) (err error) {
 	return errors.WithMessage(err, "resolve app")
 }
 
+// OptAppAndDataEnc makes an optional pair of App definition and Data encodable.
+type OptAppAndDataEnc struct {
+	App  App
+	Data Data
+}
+
+// Encode encodes an optional pair of App definition and Data.
+func (o OptAppAndDataEnc) Encode(w io.Writer) error {
+	return perunio.Encode(w, OptAppEnc{App: o.App}, o.Data)
+}
+
+// OptAppAndDataDec makes an optional pair of App definition and Data decodable.
+type OptAppAndDataDec struct {
+	App  *App
+	Data *Data
+}
+
+// Decode decodes an optional pair of App definition and Data.
+func (o OptAppAndDataDec) Decode(r io.Reader) (err error) {
+	if err = perunio.Decode(r, OptAppDec{App: o.App}); err != nil {
```

Comment:
Where is the optional character of this decoding method?

Reply:
See https://github.com/hyperledger-labs/go-perun/pull/272#discussion_r763713098